Mathlib/Algebra/BigOperators/Multiset/Basic.lean

```diff
@@ -406,6 +406,14 @@ theorem prod_map_le_prod_map {s : Multiset ι} (f : ι → α) (g : ι → α) (
 #align multiset.prod_map_le_prod_map Multiset.prod_map_le_prod_map
 #align multiset.sum_map_le_sum_map Multiset.sum_map_le_sum_map
 
+@[to_additive]
+theorem prod_map_lt_prod_map {s : Multiset ι} {f g : ι → α} [CovariantClass α α (· * ·) (· < ·)]
```

Comment:
@YaelDillies, can you confirm this assumption combined with `OrderedCommMonoid` is sensible?

Reply:
It's not. You should use `StrictOrderedCommMonoid` instead.

Reply:
Did you misspell it? Where is it defined?

Reply:
`LinearOrderedCommMonoid` doesn't talk about `<` does it?

Reply:
See my edit

Reply:
Also, why do you need linear ordering?

Reply:
Do you care about any concrete non-linear ordered monoid where multiplication is strictly monotone?

Reply:
No, I don't, but in the additive version I do.

Reply:
What's your concrete example?

Reply:
Tuples of rational numbers with elementwise addition.

Reply:
Then probably we should define `StrictOrderedCommMonoid` to mean `OrderedCommMonoid` + `CovariantClass α α (· * ·) (· < ·)`.

Reply:
OK, I'll open another PR then.

Reply:
Note for future reviewers:
`LinearOrderedCommMonoid` will not be used because it is neither necessary nor sufficient.

Reply:
@YaelDillies: how about
```suggestion
-- TODO: define and use `StrictOrderedCommMonoid`
theorem prod_map_lt_prod_map {s : Multiset ι} {f g : ι → α} [CovariantClass α α (· * ·) (· < ·)]
```

Reply:
In this case `List.prod_lt_prod'` should get the same TODO.

Reply:
And actually many lemmas from `Mathlib.Data.List.BigOperators.Basic` could be refactored to not use `CovariantClass` mixins.

Reply:
Actually I don't think that tuples of rational numbers justify `StrictOrderedCommMonoid` as we could use `OrderedCancelCommMonoid` instead.

Reply:
Agreed. I think you should just use `OrderedCancelCommMonoid`.